### PR TITLE
Memcached via Unix Domain Sockets + Basic Client impl.

### PR DESCRIPTION
**Pull request by nickamorim**
Start Memcached in UDS mode: `memcached -d -s /var/run/memcached/memcached.sock`

### CODE REVIEW

File: rust/Cargo.toml [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 rand = { version = "0.8.5", features = ["small_rng"] }
+
+[profile.release]
+debug = 1
+
+[lib]
+name = "client"
+path = "src/client.rs"
+
+[[bin]]
```
Comment (inf-rno) [inline rust/Cargo.toml:18]: does this need two brackkeys?

File: rust/Cargo.toml [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
 [dependencies]
 memcache = "0.17.0"
 memcached-rs = "0.4.2"
 rand = { version = "0.8.5", features = ["small_rng"] }
+
+[profile.release]
+debug = 1
```
Comment (inf-rno) [inline rust/Cargo.toml]: why not debug 2?

File: rust/src/client.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+impl Client {
+    pub fn connect(socket_addr: &str) -> std::io::Result<Self> {
+        let stream = UnixStream::connect(socket_addr)?;
+        Ok(Self { stream })
+    }
+
+    pub fn set(
+        &mut self,
```
Comment (inf-rno) [inline rust/src/client.rs:51]: guessing need to be mut cuz stream write needs it?

File: rust/src/task.rs [hunk 4]
```diff
@@ -103,3 +109,42 @@ impl Task for RSMem {
         TaskResult(op.into(), start.elapsed())
     }
 }
+
+struct LOCAL {
```
Comment (inf-rno) [inline rust/src/task.rs]: all caps means srs business

File: rust/Cargo.toml (hunk 2 above)
Comment (nickamorim) [reply rust/Cargo.toml]: Changed to `debug = 2` in latest commit

File: rust/Cargo.toml (hunk 1 above)
Comment (nickamorim) [reply rust/Cargo.toml:18]: I don't think it's _required_ but it's a convention I've seen [[[docs](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#configuring-a-target)]]

File: rust/src/client.rs (hunk 3 above)
Comment (nickamorim) [reply rust/src/client.rs:51]: Yep

File: rust/src/task.rs (hunk 4 above)
Comment (nickamorim) [reply rust/src/task.rs]: Yessir lol. I'll change to `Local` tho because it looks kinda weird

File: rust/Cargo.toml (hunk 1 above)
Comment (inf-rno) [reply rust/Cargo.toml:18]: oh in the doc it says that its https://toml.io/en/v1.0.0-rc.3#array-of-tables which means you can have multiple bins but only one lib

**Review by inf-rno — APPROVED** [review]

**Pull request merged**